src/internal/m365/onedrive_test.go

```diff
@@ -313,70 +167,44 @@ func TestOneDriveNightlySuite(t *testing.T) {
 }
 
 func (suite *OneDriveNightlySuite) SetupSuite() {
-	t := suite.T()
-
-	ctx, flush := tester.NewContext(t)
-	defer flush()
-
-	si := NewSuiteInfoImpl(t, ctx, tconfig.M365UserID(t), path.OneDriveService)
-
-	user, err := si.controller.AC.Users().GetByID(ctx, si.user, api.CallConfig{})
-	require.NoError(t, err, "fetching user", si.user, clues.ToCore(err))
-	si.userID = ptr.Val(user.GetId())
-
-	secondaryUser, err := si.controller.AC.Users().GetByID(ctx, si.secondaryUser, api.CallConfig{})
-	require.NoError(t, err, "fetching user", si.secondaryUser, clues.ToCore(err))
-	si.secondaryUserID = ptr.Val(secondaryUser.GetId())
-
-	tertiaryUser, err := si.controller.AC.Users().GetByID(ctx, si.tertiaryUser, api.CallConfig{})
-	require.NoError(t, err, "fetching user", si.tertiaryUser, clues.ToCore(err))
-	si.tertiaryUserID = ptr.Val(tertiaryUser.GetId())
-
-	suite.suiteInfo = si
+	suite.m365 = its.GetM365(suite.T())
+	suite.rs = its.NewResourceService(suite.m365.User, path.OneDriveService)
 }
 
 func (suite *OneDriveNightlySuite) TestRestoreAndBackup_MultipleFilesAndFolders_NoPermissions() {
-	testRestoreAndBackupMultipleFilesAndFoldersNoPermissions(suite, 0)
+	testRestoreAndBackupMultipleFilesAndFoldersNoPermissions(suite, suite.m365, suite.rs, 0)
 }
 
 func (suite *OneDriveNightlySuite) TestPermissionsRestoreAndBackup() {
-	testPermissionsRestoreAndBackup(suite, version.OneDrive1DataAndMetaFiles)
+	testPermissionsRestoreAndBackup(suite, suite.m365, suite.rs, version.OneDrive1DataAndMetaFiles)
 }
 
 func (suite *OneDriveNightlySuite) TestRestoreNoPermissionsAndBackup() {
-	testRestoreNoPermissionsAndBackup(suite, version.OneDrive1DataAndMetaFiles)
+	testRestoreNoPermissionsAndBackup(suite, suite.m365, suite.rs, version.OneDrive1DataAndMetaFiles)
 }
 
 func (suite *OneDriveNightlySuite) TestPermissionsInheritanceRestoreAndBackup() {
 	// No reason why it couldn't work with previous versions, but this is when it got introduced.
-	testPermissionsInheritanceRestoreAndBackup(suite, version.OneDrive4DirIncludesPermissions)
+	testPermissionsInheritanceRestoreAndBackup(suite, suite.m365, suite.rs, version.OneDrive4DirIncludesPermissions)
 }
 
 func (suite *OneDriveNightlySuite) TestLinkSharesInheritanceRestoreAndBackup() {
-	testLinkSharesInheritanceRestoreAndBackup(suite, version.Backup)
+	testLinkSharesInheritanceRestoreAndBackup(suite, suite.m365, suite.rs, version.Backup)
 }
 
 func (suite *OneDriveNightlySuite) TestRestoreFolderNamedFolderRegression() {
 	// No reason why it couldn't work with previous versions, but this is when it got introduced.
-	testRestoreFolderNamedFolderRegression(suite, version.All8MigrateUserPNToID)
+	testRestoreFolderNamedFolderRegression(suite, suite.m365, suite.rs, version.All8MigrateUserPNToID)
 }
 
 func testRestoreAndBackupMultipleFilesAndFoldersNoPermissions(
-	suite oneDriveSuite,
+	suite tester.Suite,
+	m365 its.M365IntgTestSetup,
+	irs its.ResourceServicer,
```

Comment:
nit: could you please use a more descriptive name for this parameter? Ditto for other functions that are similar to this one